npm/src/s3upload/tdr-fetch-handler.ts

```diff
@@ -0,0 +1,126 @@
+import { HttpHandler, HttpRequest, HttpResponse } from "@aws-sdk/protocol-http"
+import { HttpHandlerOptions, HeaderBag } from "@aws-sdk/types"
+
+function requestTimeout(timeoutInMs = 0): Promise<never> {
+  return new Promise((resolve, reject) => {
+    if (timeoutInMs) {
+      setTimeout(() => {
+        const timeoutError = new Error(
+          `Request did not complete within ${timeoutInMs} ms`
+        )
+        timeoutError.name = "TimeoutError"
+        reject(timeoutError)
+      }, timeoutInMs)
+    }
+  })
+}
+
+export interface FetchHttpHandlerOptions {
+  /**
+   * The number of milliseconds a request can take before being automatically
+   * terminated.
+   */
+  requestTimeout?: number
+}
+
+/**
+ * This is copied mostly from https://github.com/aws/aws-sdk-js-v3/blob/main/packages/fetch-http-handler/src/fetch-http-handler.ts
+ * There are a few changes
+ * We set credentials to 'include' so the cookies are sent
+ * We stop sending query parameters as they're not needed.
+ * We remove the bucket name from the path
+ */
+export class TdrFetchHandler implements HttpHandler {
+  private readonly requestTimeout?: number
+
+  constructor({ requestTimeout }: FetchHttpHandlerOptions = {}) {
+    this.requestTimeout = requestTimeout
+  }
+
+  destroy(): void {
+    // Do nothing. TLS and HTTP/2 connection pooling is handled by the browser.
+  }
+
+  handle(
+    request: HttpRequest,
+    { abortSignal }: HttpHandlerOptions = {}
+  ): Promise<{ response: HttpResponse }> {
+    const requestTimeoutInMs = this.requestTimeout
+
+    // if the request was already aborted, prevent doing extra work
+    if (abortSignal?.aborted) {
+      const abortError = new Error("Request aborted")
+      abortError.name = "AbortError"
+      return Promise.reject(abortError)
+    }
+
+    let path = `/${request.path.split("/").slice(2).join("/")}`
+
+    const { port, method } = request
+    const url = `${request.protocol}//${request.hostname}${
+      port ? `:${port}` : ""
+    }${path}`
+    // Request constructor doesn't allow GET/HEAD request with body
+    // ref: https://github.com/whatwg/fetch/issues/551
+    const body =
+      method === "GET" || method === "HEAD" ? undefined : request.body
+
+    const requestOptions: RequestInit = {//eslint-disable-line
```

Comment:
Is this comment `//eslint-disable-line` needed?

Reply:
The RequestInit is part of the base typescript dom library so it doesn't need an explicit import which eslint doesn't like. I had a google though and you can add it to the .eslintrc and get rid of the comment so I've done that.